### ISSUE TITLE
Notebook web UI should indicate when dependency resolution is happening, and when it fails.

### ISSUE THREAD

**Issue by ryan-williams** [andypetrella/spark-notebook#321]
This is in the same vein as #294 but different enough to warrant its own issue, I think.

When a notebook fails to successfully resolve sbt/Maven dependencies, no good indication is given in the web UI, and the cause is hard to diagnose.

For example, one of my teammates started a notebook pointing at his local Maven repo in his homedir. However, our server is running as me, and doesn't have write perms to that directory.

The result is that the *server output*, not even the notebook's log file, has a bunch of info about Maven dependencies failing to resolve. [Here is a gist of the relevant server stdout](https://gist.github.com/ryan-williams/4c61434957eec27d32f7). Lots of `[warn]` messages, and then bizarre silence at the end.

Notebooks should indicate in the web UI when they are in the process of resolving Maven deps, and they should indicate errors in the web UI when this process fails.

**Comment by andypetrella** [comment]
indeed, this is crappy, and should almost be resolved by having a panel showing the logs.
Then a notification for it would be appreciated (using the notification are on the upper right for instance)

**Comment by andypetrella** [comment]
We're getting there with https://github.com/andypetrella/spark-notebook/pull/334

Now, everything is logged in the browser, so we just need a bit of UI telling more about this.

The deps are one problem among others that can only be discovered using the logs atm, including #294.

I guess we'll need something more sofisticated than a panel/console with logs, but it could also be delayed to leave the place to other features I guess.

**Comment by ryan-williams** [comment]
Cool, thanks for the progress here @andypetrella. Having console logs is a great start; putting the info in a simple panel in the actual page would be good for e.g. searchability and general unification of the UI, I think, but I'll try out the current version and let you know what I think.

**Comment by andypetrella** [comment]
hey @ryan-williams no problemo, this was annoying me as much as you ^^.

I'm using it a lot now, and I find it really helpful to be honest. And actually, the browser console is fine so far because it's well managed by the browser (in term of memory, cpu), you can also search, enable, disable, filter. Actually, if a panel is added to the UI, it's gonna be the same ^^ (some limited stuffs exists already: https://github.com/bahmutov/console-log-div).

let me know what you think and what you'd expect :-D

**Comment by ryan-williams** [comment]
Cool, I'll let you know when I get a chance to try this out, but feel free to close if it feels done to you, your explanation makes sense to me.

**Comment by ryan-williams** [comment]
OK I just tried this out; I saw this in my console as soon as the notebook opened:

![](http://cl.ly/image/3o1X1y2k0a1g/Screen%20Shot%202015-10-08%20at%202.06.53%20PM.png)

However, I could see in the server logs that dep resolution was still happening, and no YARN app existed yet that backed my notebook, so I'm not clear on which logs I'm viewing in my console.

**Comment by ryan-williams** [comment]
Ah, turns out I was running my old SN version. New version shows the logs, looks great, thanks @andypetrella !

**Comment by andypetrella** [comment]
Aaaaaah I was frightened yesterday when I saw that, but I was too screwed to be able to answer anything so I delayed it.... ^^
That's cool you like it (and it works)!

**Comment by huitseeker** [comment]
@andypetrella perhaps this deserves a close, and #294 might deserve a look-over
